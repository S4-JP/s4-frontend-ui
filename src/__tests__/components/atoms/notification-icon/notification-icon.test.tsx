import React from 'react'
import { render } from '@testing-library/react'
import { NotificationIcon, Color } from '../../../../index'

describe('notification-icon のレンダリング確認', () => {
  test('オプションなしで設定した場合、初期値でレンダリングされる。', () => {
    const { getByTestId } = render(<NotificationIcon number={10} />)

    expect(getByTestId('atoms-ni-wrap')).toMatchSnapshot()

    expect(getByTestId('atoms-ni-wrap')).toHaveStyle(
      `background-color: ${Color.COMPONENT.NOTICE};`
    )
    expect(getByTestId('atoms-ni-wrap').textContent).toBe('10')
  })

  test('bgColor=#000を設定した場合、設定値でレンダリングされる。', () => {
    const { getByTestId } = render(
      <NotificationIcon number={10} bgColor="#000" />
    )

    expect(getByTestId('atoms-ni-wrap')).toHaveStyle('background-color: #000;')
  })

  test('omit=trueを設定した場合、設定値でレンダリングされる。', () => {
    const { getByTestId } = render(
      <NotificationIcon number={10} bgColor="#000" omit={true} />
    )

    expect(getByTestId('atoms-ni-wrap')).toHaveStyle(
      'width: 8px;height: 8px;margin: -12px -6px 0 0;'
    )
    expect(getByTestId('atoms-ni-wrap')).toHaveStyle(
      'width: 8px;height: 8px;margin: -12px -6px 0 0;'
    )
  })
})
