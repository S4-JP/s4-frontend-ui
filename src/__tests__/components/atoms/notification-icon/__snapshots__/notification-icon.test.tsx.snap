// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notification-icon のレンダリング確認 オプションなしで設定した場合、初期値でレンダリングされる。 1`] = `
<div
  class="sc-TRNrF eUpOFA"
  data-testid="atoms-ni-wrap"
>
  10
</div>
`;
